discord/ui/item.py

```diff
@@ -42,11 +52,39 @@
     from .modal import BaseModal
     from .view import BaseView
 
-I = TypeVar("I", bound="Item")
+I = TypeVar("I", bound="Item", covariant=True)
+
 T = TypeVar("T", bound="ItemInterface", covariant=True)
 V = TypeVar("V", bound="BaseView", covariant=True)
 M = TypeVar("M", bound="BaseModal", covariant=True)
+
+V_co = TypeVar("V_co", bound="BaseView", covariant=True)
+
 ItemCallbackType = Callable[[Any, I, Interaction], Coroutine[Any, Any, Any]]
+SetItemCallbackType = (
+    Callable[[Interaction], Coroutine[object, Any, Any]]
+    | Callable[[Interaction, I], Coroutine[object, Any, Any]]
+    | Callable[[Interaction, I, V], Coroutine[object, Any, Any]]
+)
+
+
+class _ProxyItemCallback:
+    def __init__(
+        self, func: SetItemCallbackType, item: ViewItem, parameters_amount: int
+    ) -> None:
+        self.func: SetItemCallbackType = func
+        self.item: ViewItem = item
+        self.parameters_amount: int = parameters_amount
+
+    def __call__(self, interaction: Interaction) -> Coroutine[Any, Any, Any]:
+        if self.parameters_amount == 1:
+            return self.func(interaction)  # type: ignore # type checker doesn't like optional params
+        elif self.parameters_amount == 2:
+            return self.func(interaction, self.item)  # type: ignore # type checker doesn't like optional params
+        elif self.parameters_amount == 3:
+            return self.func(interaction, self.item, self.item.view)  # type: ignore # type checker doesn't like optional params
```

Comment:
Shouldn't view be first ? It is most likely representing "self" in this case

Reply:
AFAIK, more people want access to the button/item rather than the view, so they can do `(interaction, button)` instead of having to do `(interaction, view, button)`.